sweetest/src/main/java/com/mysugr/sweetest/framework/coroutine/CoroutinesTestContext.kt

```diff
@@ -0,0 +1,23 @@
+package com.mysugr.sweetest.framework.coroutine
+
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.newSingleThreadContext
+import kotlin.coroutines.CoroutineContext
+
+class CoroutinesTestContext {
+    private val name = CoroutineName("testCoroutine${instanceCounter++}")
+    private val supervisorJob = SupervisorJob()
+    val coroutineContext: CoroutineContext
+        get() = coroutineDispatcher + supervisorJob + name
+
+    suspend fun testFinished() {
+        supervisorJob.cancelAndJoin()
+    }
+
+    companion object {
+        val coroutineDispatcher by lazy { newSingleThreadContext("testDispatcher") }
```

Comment:
newSingleThreadContext will be removed in the future:
https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/new-single-thread-context.html 
In order for this not to break based on changes in the by the Kotlin team i suggest something like: 
`val coroutineDispatcher by lazy { Executors.newSingleThreadExecutor().asCoroutineDispatcher() }`